src/corehost/build.cmd

```diff
@@ -34,6 +34,7 @@ if /i [%1] == [rid]         ( set __TargetRid=%2&&shift&&shift&goto Arg_Loop)
 if /i [%1] == [toolsetDir]  ( set "__ToolsetDir=%2"&&shift&&shift&goto Arg_Loop)
 if /i [%1] == [hostver]     (set __HostVersion=%2&&shift&&shift&goto Arg_Loop)
 if /i [%1] == [apphostver]  (set __AppHostVersion=%2&&shift&&shift&goto Arg_Loop)
+if /i [%1] == [comhostver]  (set __ComHostVersion=%2&&shift&&shift&goto Arg_Loop)
```

Comment:
Does this argument be passed in somewhere?

I am wondering why we have so many different versions - why can't all hosts have same version?

Reply:
That is a great question. I would like to do that, but alas the entire hosting design is an amalgam of various designs and considerations that don't seem to have a cohesive design. I could change it _but_ this is plumbed through the build and deployment mechanisms so a bunch would have to change. I am more than willing to do that though.

@vitek-karas Any concerns you can think of? Are you planning on doing this?

Reply:
My understanding is that the versions are just a symptom of a different intent. I think we use the versions to basically "mark" the components which we want to update in a servicing release.
(I think, but I'm not sure, that if the version didn't change, we don't ship/update the binary again).

This is super important for `dotnet.exe` - updating this binary requires a reboot and that is something we try to avoid at almost all costs. Basically any servicing release will not update this, and even feature releases may decide to not update it either.

As for the other components I don't really know why we version them separately. I think this goes more to the setup team, as they consume these/version these.
My take:
* `hostfxr` - should simply build the latest version always - and ship it. At runtime we always pick the latest available. The only "tricky" part are preview releases, where in theory preview might have bugs, but any invocation will pick up the latest (which might be preview). But this is a behavior we already shipped with many times.
* `hostpolicy` - this ships with the runtime, should be versioned with the runtime and so on. We do ship it as a separate Nuget package, but that is probably just a historical thing due to it living in a different repo from the runtime. Otherwise this library "belongs" to CoreCLR.
* `apphost` - SDK uses this to publish apps (and now even to build them). It needs to be packaged separately since it's a component which should not be part of the runtime package which is installed on machines. That said it should version with the runtime, no reason not to do that. SDK should pick the matching version for whatever runtime it picked to publish the app for.
* `comhost` - same story as `apphost`. For this change I think we should make the `comhost` and `apphost` versions the same. In fact I would expect we ship them in the same NuGet package - which leads to the question on how are we going to ship it - I assume the same as `apphost` since it's again a component consumed only by SDK.

/cc @rakeshsinghranchi  for the setup related topics

Reply:
Will do this now that I am merging the entire COM host work into this PR.